packages/cloud_firestore/cloud_firestore/example/test_driver/document_reference_e2e.dart

```diff
@@ -46,6 +46,28 @@ void runDocumentReferenceTests() {
         }, count: 1, reason: "Stream should only have been called once."));
       });
 
+      test('listens to multiple documents', () async {
+        DocumentReference doc1 = await initializeTest('document-snapshot-1');
+        DocumentReference doc2 = await initializeTest('document-snapshot-2');
+
+        await doc1.set({'test': 'value1'});
+        await doc2.set({'test': 'value2'});
+
+        Stream<DocumentSnapshot> stream1 = doc1.snapshots();
+        Stream<DocumentSnapshot> stream2 = doc2.snapshots();
+
+        final completer1 = Completer<String>();
+        final completer2 = Completer<String>();
+
+        stream1.listen((s) => completer1.complete(s.data()['test']));
+        stream2.listen((s) => completer2.complete(s.data()['test']));
+
+        expect(
+          await Future.wait([completer1.future, completer2.future]),
+          ['value1', 'value2'],
+        );
```

Comment:
nit but I'd prefer:

```dart
expectLater(completer1.future, 'value1');
expectLater(completer2.future, 'value2');
```

Reply:
Alternatively we could do:

```dart
final value1 = doc1.snapshots().first.then((s) => s.data()['test']);
final value2 = doc2.snapshots().first.then((s) => s.data()['test']);

await expectLater(value1, completion('value1'));
await expectLater(value2, completion('value2'));
```

Reply:
If we make this change, the other tests added could do the same

Reply:
I like the second approach, much more concise.